src/main/kotlin/org/wa/rceditor/api/RCManagerImpl.kt

```diff
@@ -0,0 +1,40 @@
+package org.wa.rceditor.api
+
+import io.reactivex.Completable
+import io.reactivex.Maybe
+import org.wa.rceditor.domain.RCManager
+import org.wycliffeassociates.resourcecontainer.ResourceContainer
+import org.wycliffeassociates.resourcecontainer.entity.Checking
+import org.wycliffeassociates.resourcecontainer.entity.DublinCore
+import org.wycliffeassociates.resourcecontainer.entity.Manifest
+import tornadofx.*
+
+class RCManagerImpl: RCManager {
```

Comment:
Instead of a vague "manager" name, something like "RCWriter" and "RCReader" or "ResourceContainerIO"

Reply:
I prefer IO since this class does both, but you could split reading and writing into two classes.

But to further add to what Matthew is saying; yes. Names like "Manager" tells us nothing about the class and its intent. What it tells us is that it is likely a "Controller" of some sort, but even the name "Controller" is useless other than to tell us it isn't a View. But you can read the methods and member variables and quickly identify that.

Naming is hard, but this one especially is an anti-pattern. Managers indicate that a class is probably doing too many things (breaking the single responsibility principle), or eventually will be. Even if it's not, it indicates an uncertainty in the code on the behalf of the developer, and thus possibly a lack of understanding of the problem attempting to be "solved."